### ISSUE TITLE
Wrong class in CDI Extensions Check

### ISSUE THREAD

**Issue by stefanorg** [derjust/spring-data-dynamodb#228]
Hi i think that this: 
https://github.com/derjust/spring-data-dynamodb/blob/40bd99e9180d61450d371bc13e8da08fb4e94edd/src/main/java/org/socialsignin/spring/data/dynamodb/repository/cdi/DynamoDBRepositoryExtension.java#L76

should be

```java
... && AmazonDynamoDB.class ...
```

This way in a CDI context i can have a Producer of AmazonDynamoDB:

```java
@Bean @Produces
    public AmazonDynamoDB amazonDynamoDB() {
//        return AmazonDynamoDBClientBuilder
//          .standard()
//          .withCredentials(amazonAWSCredentialsProvider())
//          .withRegion(Regions.EU_WEST_1).build();

        return AmazonDaxClientBuilder
          .standard()
          .withCredentials(amazonAWSCredentialsProvider())
          .withRegion(Regions.EU_WEST_1)
          .withEndpointConfiguration("**********************************")
          .build();


    }
``` 
Thanks

**Comment by derjust** [comment]
Yes, resorting to the interface is all we need - no need to be that 'specific'

Will be part of `5.1.0` (Available as part of `5.1.0-SNAPSHOT` as of now)

**Comment by stefanorg** [comment]
Hi i'm not able to see this fix in 5.1.0 branch.

https://github.com/derjust/spring-data-dynamodb/blob/v5.1.0/src/main/java/org/socialsignin/spring/data/dynamodb/repository/cdi/DynamoDBRepositoryExtension.java

Do you need a pull request for this? 
When do you think this can be fixed?

Thank for your work :+1:
